pkg/kubectl/scheme/scheme.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Registry is an instance of an API registry.  This is an interim step to start removing the idea of a global
+// API registry.
+var Registry = registered.NewOrDie(os.Getenv("KUBE_API_VERSIONS"))
+
+// Scheme is the default instance of runtime.Scheme to which types in the Kubernetes API are already registered.
+var Scheme = runtime.NewScheme()
```

Comment:
@deads2k @sttts 
This PR wants to make kubectl to use its own instance of scheme instead of [the global scheme](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/legacyscheme/scheme.go). This work is important for moving kubectl out of the core.
All generic operations in this PR works well, e.g. kubectl create|delete -f.
But the type specific commands (e.g. kubectl create namespace|deployment) don't work unless I install the types like in https://github.com/kubernetes/kubernetes/blob/master/pkg/api/install/install.go#L32.

Otherwise, I will hit the following error.
```
the provided version "v1" has no relevant versions: group  has not been registered
no matches for /, Kind=Namespace
```

I want to know what is the right solution(short term and long term).

Reply:
cc @caesarxuchao

Reply:
Very good point. The `Install` funcs actually work for your use-case. But we have to get rid of the `init()` funcs and move that somewhere else. Actually, maybe we can add that to pkg/api/legacyscheme. @deads2k any opinion?

Reply:
We can do this in a follow-up, no need here in the PR.

Reply:
Actually, when you have moved to external types, the install funcs are not the right way to register types, but compare k8s.io/client-go/kubernetes/scheme/register.go.

Reply:
kubectl should eventually not depend on internal types, which means it eventually should not use the install func in `pkg/api` and etc.

Reply:
Did you find the root cause of the error message? IIUC, you moved to the `k8s.io/client-go/kubernetes/scheme/register.go` pattern and then you saw the error. But do you know what's the delta between the `Install` and `k8s.io/client-go/kubernetes/scheme/register.go` pattern that causes the error?

Reply:
I'm 80% sure it is because of not registering the internal types: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/install/install.go#L41

Reply:
To make the PR tests pass, I will call all install funcs in kubectl's scheme (to install the internal types I guess).

After switching all kubectl commands to use external types, hopefully we won't need to install internal types any more.

Reply:
@deads2k I'd love to hear your thoughts!

Reply:
Another maybe related thought: the `Registry` is only used via the `install` pattern. Hence, if you rely on statically registered API groups for the RESTMapper, then this won't work without using `install`. In fact, the internal types are not necessary, but you cannot get one without the other (which makes me wonder how we did that in client-go when we were still using the Registry there).